presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestDistributedQueries.java

```diff
@@ -714,17 +721,42 @@ public void testQueryLoggingCount()
     {
         QueryManager queryManager = ((DistributedQueryRunner) getQueryRunner()).getCoordinator().getQueryManager();
         executeExclusively(() -> {
+            assertTrueWithTimeout(
+                    () -> queryManager.getAllQueryInfo().stream().allMatch(QueryInfo::isFinalQueryInfo),
+                    new Duration(1, MINUTES),
+                    "There are still queries in process: %s", queryManager.getAllQueryInfo());
+
             long beforeQueryCount = queryManager.getStats().getCompletedQueries().getTotalCount();
             assertUpdate("CREATE TABLE test_query_logging_count AS SELECT 1 foo_1, 2 foo_2_4", 1);
             assertQuery("SELECT foo_1, foo_2_4 FROM test_query_logging_count", "SELECT 1, 2");
             assertUpdate("DROP TABLE test_query_logging_count");
             assertQueryFails("SELECT * FROM test_query_logging_count", ".*Table .* does not exist");
-
-            long queryCount = queryManager.getStats().getCompletedQueries().getTotalCount() - beforeQueryCount;
-            assertEquals(queryCount, 4, format("expected [%s] but found [%s]", 4, queryCount));
+            // TODO: Figure out a better way of synchronization
+            assertTrueWithTimeout(
+                    () -> queryManager.getStats().getCompletedQueries().getTotalCount() - beforeQueryCount == 4,
+                    new Duration(1, MINUTES),
+                    "expected [%s] but found [%s]", 4, queryManager.getStats().getCompletedQueries().getTotalCount() - beforeQueryCount);
         });
     }
 
+    private static void assertTrueWithTimeout(BooleanSupplier condition, Duration timeout, String message, Object... args)
```

Comment:
Consider naming `assertEventually` (just like in "eventual consistency")

There are also hand-coded busy waits in tests elsewhere, e.g. https://github.com/prestodb/presto/pull/7677/files#diff-12e16468ce25304298fd890d92f2fea5R258, f472567bacfef9423f45706b187f523c19dbac51. Maybe we could have some reusable method implemented once?